### PR TITLE
Do not raise warning when there is a working intersphinx inventory

### PR DESCRIPTION
**Pull request by bsipocz**
This PR should fix the issue we see with various astropy packages when the scipy server is down, making the default intersphinx inventory inaccessible. However we host another copy in our server and thus the warnings about the missing default inventories shouldn't fail our docs builds.

https://github.com/astropy/astropy-helpers/issues/327

and travis example: https://travis-ci.org/astropy/ccdproc/jobs/255026568#L677

### Feature or Bugfix
- Bugfix

### CODE REVIEW

**Comment by bsipocz** [comment]
@tk0miya @shimizukawa - could you please review this and let me know what you think? I'm open to any alternative solution, my only aim is to get rid of those warnings that unnecessary fail our builds.

File: sphinx/ext/intersphinx.py [hunk 1]
```diff
@@ -241,12 +243,28 @@ def load_mappings(app):
...
+                    continue
+
                 if invdata:
                     inventories.cache[uri] = (name, now, invdata)
                     update = True
                     break
 
+        if len(failures) < len(invs):
+            logger.info("encountered some issues with some of the inventories,"
+                        " but they had working alternatives:")
```
Comment (tk0miya) [inline sphinx/ext/intersphinx.py:259]: what does "alternatives" mean here? where can I set up them?

Comment (bsipocz) [reply sphinx/ext/intersphinx.py:259]: I meant this: https://github.com/sphinx-doc/sphinx/issues/1537, usage is e.g. here:
https://github.com/astropy/astropy-helpers/blob/b5b722345db475d91edb9fbae41d3d0364d6764a/astropy_helpers/sphinx/conf.py#L51

Comment (tk0miya) [reply sphinx/ext/intersphinx.py:259]: Thanks, I understand. I don't know such feature. I looks good.

**Review by tk0miya — CHANGES_REQUESTED** [review]
Could you add testcase for this please?

**Comment by bsipocz** [comment]
@tk0miya - Can we merge this at some point? I would like to avoid missing 1.7. Or since this actually address a bug, maybe 1.6.5?

**Comment by tk0miya** [comment]
Yes, I'd like to merge this. Could you add testcase for this please? Then I will merge this soon.
In addition, this is absolutely bugfix. so could you rebase this to stable branch?

**Comment by tk0miya** [comment]
ping. any updates?

**Comment by tk0miya** [comment]
Finally I merge this manually (with minor fix. see #4528).
Thank you for your contribution!

**Comment by bsipocz** [comment]
@tk0miya - I'm sorry for dropping the ball on this, and thanks for finishing it up and merging.

**Comment by tk0miya** [comment]
not at all :-)

**Pull request closed without merging**